modules/iam-assumable-role/variables.tf

```diff
@@ -16,6 +16,16 @@ variable "trusted_role_services" {
   default     = []
 }
 
+variable "trust_policy_conditions" {
+  description = "[Condition constraints](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document#condition) applied to the trust policy"
+  type = list(object({
```

Comment:
I also think we should start introducing such stricter types for variables where it makes sense. 👍 

In my experience with other modules (e.g., S3 bucket), 100% concrete variable types are impossible, which is unfortunate. We need to be careful about that. :)